hnn_core/tests/test_io.py

```diff
@@ -1,26 +1,21 @@
-import os.path as op
+from pathlib import Path
 from numpy.testing import assert_allclose
 from h5io import write_hdf5
 import pytest
 
-import hnn_core
-from hnn_core import read_network
-from hnn_core import simulate_dipole
-from hnn_core import read_params
-from hnn_core import jones_2009_model, law_2021_model, calcium_model
+from hnn_core import (read_network, simulate_dipole, read_params,
+                      jones_2009_model, law_2021_model, calcium_model)
 
-hnn_core_root = op.dirname(hnn_core.__file__)
-params_fname = op.join(hnn_core_root, 'param', 'default.json')
+hnn_core_root = Path(__file__).parents[1]
 
 
 @pytest.mark.parametrize("network_model",
                          [law_2021_model, calcium_model,
                           jones_2009_model])
 def test_network_io(tmp_path, network_model):
     # For simulation to be shorter(Discuss)
-    params = op.join(hnn_core_root, 'param', 'default.json')
-    if isinstance(params, str):
-        params = read_params(params)
+    params_path = Path(hnn_core_root, 'param', 'default.json')
+    params = read_params(params_path)
     params['N_pyr_x'] = 3
```

Comment:
@gtdang the problem is here btw! At the moment we don't have a real API that let's us change the number of cells in the network, this is just a hack in the tests because smaller networks simulate much faster.

Essentially the saved network is a 3x3 grid, but then the loaded network is the default 10x10 grid.

One way this could be remedied is allowing users to pass `net  = Network(..., n_pyr_x=3, n_pyr_y=3)` explicitly so that the network size is saved in the HDF5 file.

@rythorpe @jasmainak any thoughts?

Reply:
For context @rythorpe and @jasmainak, @gtdang found that the `net.connectivity` changed after writing to file and reloading (it's changing because the networks are of different sizes)

Reply:
I believe the last point is the most accurate. All of the functions to setup the cell positions and grid size are called right when the network is initialized...

Reply:
@ntolley you're advocating for an API change here right? If so, I'd definitely support that. My vote would be for a single argument like `net = Network(..., mesh_shape=(3, 3))`. This is nitpicky, but my use of "mesh" instead of "grid" comes from the desire to support more flexible network cell positioning functions that e.g. place cells in hexogonal patterns or random patterns on a 2D plane rather than on a rectangular grid.

Reply:
+1 of making this explicit in our API. Does this apply only to pyramidal cells or basket cells as well? Do we need a dict to specify the cell type?

Reply:
Just discussed with @gtdang, we're thinking that an intermediate solution will be to implement `getter` and `setter` decorators for these variables.

I think making and API through the network `__init__()` will be worthwhile in the future, but we might want to discuss these details (like a specified mesh) in a separate PR

Reply:
@gtdang actually I just realized this won't work because `net.connectivity` is populated when we call `jones_2009_model()`. So this would actually require having a `mesh_shape()` argument for all of the implemented models, and then pass that to `Network`.

It's not the prettiest solution but I'll try and give it a shot.